src/Survey/index.jsx

```diff
@@ -0,0 +1,166 @@
+import "./survey.css";
+import titleCard from "../Assets/redBanner.png";
+import { Button } from "../Components/Buttons/Button";
+import { Radio } from "../Components/RadioButton/Radio";
+import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import useLocalStorage from "../Components/UseLocalStorage";
+import { Formik, Form, Field, ErrorMessage, FieldArray } from "formik";
+import * as Yup from "yup";
+
+const Survey = () => {
+  const [isDisabled, setIsDisabled] = useState(false);
+  const [name, setName] = useState("");
+  const [age, setAge] = useState(0);
+  const [email, setEmail] = useState("");
+  const [gender, setGender] = useState("Male");
+  const [submitSuccess, setSubmitSuccess] = useState(false);
+
+  const initialValues = { name, age, email, gender };
+
+  const validationSchema = Yup.object({
+    name: Yup.string().required("Please enter a name").min(5, "Too Short"),
+    age: Yup.number()
+      .required("Please enter a proper age")
+      .positive("Please enter a proper age")
+      .max(100, "Please enter a proper age"),
+    email: Yup.string()
+      .email("Invalid email format!")
+      .required("Enter a valid mail ID"),
+  });
+
+  const navigate = useNavigate();
+
+  const [surveyData, setSurveyData] = useLocalStorage("Survey_Taken_By", "");
+
+  const showData = () => {
+    console.log("Name: ", name);
+    console.log("Age: ", age);
+    console.log("Gender: ", gender);
+    console.log("Email: ", email);
+  };
+
+  const onReset = () => {
+    setGender("Male");
+  };
+
+  const onSubmit = () => {
+    showData();
+
+    let survey = { name, age, email, gender };
+    setSurveyData(...surveyData, survey);
+    setSubmitSuccess(true);
```

Comment:
Why are we setting the state here? It's not required. Please use navigate in the onSubmit method itself.

Reply:
Resolved